delft/utilities/Transformer.py

```diff
@@ -0,0 +1,171 @@
+        it will try to use huggingface as fallback solution.
+        """
+        if self.loading_method == LOADING_METHOD_HUGGINGFACE_NAME:
+            self.tokenizer = AutoTokenizer.from_pretrained(self.name,
+                                                           add_special_tokens=add_special_tokens,
+                                                           max_length=max_sequence_length,
+                                                           add_prefix_space=add_prefix_space)
+
+        elif self.loading_method == LOADING_METHOD_LOCAL_MODEL_DIR:
+            self.tokenizer = AutoTokenizer.from_pretrained(self.local_dir_path,
+                                                           add_special_tokens=add_special_tokens,
+                                                           max_length=max_sequence_length,
+                                                           add_prefix_space=add_prefix_space)
+        elif self.loading_method == LOADING_METHOD_PLAIN_MODEL:
+            self.tokenizer = BertTokenizer.from_pretrained(self.local_vocab_file)
+
+        elif self.loading_method == LOADING_METHOD_DELFT_MODEL:
+            config_path = os.path.join(".", self.local_dir_path, TRANSFORMER_CONFIG_FILE_NAME)
+            self.transformer_config = AutoConfig.from_pretrained(config_path)
+            self.tokenizer = AutoTokenizer.from_pretrained(os.path.join(self.local_dir_path, DEFAULT_TRANSFORMER_TOKENIZER_DIR), config=self.transformer_config)
+
+        self.bert_preprocessor = BERTPreprocessor(self.tokenizer, empty_features_vector, empty_char_vector)
+
+        return self.bert_preprocessor
+
+    def save_tokenizer(self, output_directory):
+        self.tokenizer.save_pretrained(output_directory)
+
+    def instantiate_layer(self, load_pretrained_weights=True) -> Union[object, TFAutoModel, TFBertModel]:
+        if self.loading_method == LOADING_METHOD_HUGGINGFACE_NAME:
+            if load_pretrained_weights:
+                self.transformer_config = AutoConfig.from_pretrained(self.name)
+            else:
+                config_path = os.path.join(".", self.local_dir_path, TRANSFORMER_CONFIG_FILE_NAME)
+                self.transformer_config = AutoConfig.from_pretrained(config_path)
+            return TFAutoModel.from_config(self.transformer_config)
+
+        elif self.loading_method == LOADING_METHOD_LOCAL_MODEL_DIR:
+            if load_pretrained_weights:
+                transformer_model = TFAutoModel.from_pretrained(self.local_dir_path, from_pt=True)
+                self.transformer_config = transformer_model.config
+                return transformer_model
+            else:
+                transformer_model = TFAutoModel.from_pretrained(self.local_dir_path, from_pt=True)
+                self.transformer_config = transformer_model.config
+                return transformer_model
+
+        elif self.loading_method == LOADING_METHOD_PLAIN_MODEL:
```

Comment:
So I suppose it's currently impossible to instantiate a transformer_layer from a locally downloaded weight file, correct? 
We need to load the whole directory with standard names, but we can't with typical renaming of the config/weight files?

Reply:
I did not find a way to load the plain weight files... we might need to convert the model as a pytorch model and then load the whole directory.